content/pages/how-to/use-direct-upload-with-continuous-integration.md

```diff
@@ -0,0 +1,178 @@
+ 
+### Setup
+ 
+Create a `.circleci/config.yml` file at the root of your project. This file contains the jobs that will be executed based on the order of your workflow. In your `config.yml` file, copy the following content:
+ 
+```yaml
+---
+filename: .circleci/config.yml
+---
+version: 2.1
+jobs:
+ Publish-to-Pages:
+   docker:
+     - image: cimg/node:17.2.0
+   steps:
+     - checkout
+     # Run your project's build step
+     - run: npm install && npm run build
+     # Publish with wrangler
+     - run: npx wrangler pages publish dist --project-name=<PROJECT NAME> # Replace dist with the name of your build folder and input your project name 
+       environment:
+        CLOUDFLARE_ACCOUNT_ID: $CLOUDFLARE_ACCOUNT_ID
+        CLOUDFLARE_API_TOKEN: $CLOUDFLARE_API_TOKEN
+   
+workflows:
+ Publish-to-Pages-workflow:
+   jobs:
+     - Publish-to-Pages
+ 
+ 
+```
+ 
+Your continuous integration workflow is broken down into Jobs when using CircleCI. From the code block above, you can see that you first define a list of jobs that run on each commit. For example, your repository will run on a prebuilt docker image `cimg/node:16.7.0`. It first checks out the repository with the Node version specified in the image.
+ 
+{{<Aside type="note" header="Note">}}
+ 
+If your project uses a different Node version less than 16, you will have to upgrade your Node version as Wrangler requires at least Node.js v16.7.0.
+ 
+{{</Aside>}}
+ 
+The job then proceeds to run all the steps specified, after which you will need to define a `workflow` at the end of your file. You can learn more about creating a custom process with CircleCI from the [official documentation](https://circleci.com/docs/2.0/concepts/).
+ 
+ 
+ 
+## Using Travis CI for CI/CD
+ 
+Travis CI is an open-source continuous integration tool that handles specific tasks, such as Pull requests and code pushes for your project workflow. Travis CI can be integrated into your GitHub projects, databases, and other pre-installed services enabled in your build configuration. To use Travis CI, you should have A GitHub, Bitbucket, GitLab or Assembla account.
+
+### Add Cloudflare credentials to TravisCI
+ In your Travis project, you will need to add the Cloudflare credentials you have generated from the Cloudflare dashboard to access them in your `travis.yml` file. Head over to your Travis CI dashboard and select your current project > More options > settings > Environment variables.
+
+Set the environment variable's name and value and the branch you want it to be attached to; You can also set the privacy of the value.
+
+### Setup
+Go to Travis-ci.com and enable your repository by login in with your preferred provider. This guide uses GitHub. Next, create a `.travis.yml` file and copy the following into the file:
+
+```yaml
+---
+filename: .travis.yml
+---
+language: node_js
+node_js:
+- "16.17.0" # You can specify more versions of Node you want your CI process to support
```

Comment:
lets use same version of node across all the examples

Reply:
@eidam 18.0.0 ok?